homeassistant/helpers/entity_platform.py

```diff
@@ -272,15 +272,16 @@ def add_entities(self, new_entities, update_before_add=False):
             else:
                 config_entry_id = None
 
-            device = entity.device
-            if device is not None:
+            device_info = entity.device_info
+            if config_entry_id is not None and device_info is not None:
                 device = device_registry.async_get_or_create(
-                    connection=device['connection'],
-                    identifiers=device['identifiers'],
-                    manufacturer=device['manufacturer'],
-                    model=device['model'],
-                    name=device.get('name'),
-                    sw_version=device.get('sw_version'))
+                    config_entry=config_entry_id,
+                    connections=device_info['connections'],
```

Comment:
This will blow up if the key `connections`, `identifiers` or `manufacturer` does not exist. We should use `device_info.get('connections', [])`.

We should probably be defensive for all values and use `get`

Reply:
Yes, we already have an issue report on this it seems :)